### PR TITLE
ZK-3702: Combowidget components with autodrop trigger popup on non co…

### PR DESCRIPTION
**Pull request by klyvechen**
…ntent relevant keys

### CODE REVIEW

File: zul/src/archive/web/js/zul/inp/ComboWidget.js [hunk 1]
```diff
@@ -595,8 +595,8 @@ zul.inp.ComboWidget = zk.$extends(zul.inp.InputWidget, {
 			return;
 		}
 
-		if (keyCode == 18 || keyCode == 27 || keyCode == 13
-		|| (keyCode >= 112 && keyCode <= 123)) //ALT, ESC, Enter, Fn
+		if (keyCode == 18 || keyCode == 27 || keyCode == 13 || keyCode == 93 || keyCode == 17
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/ComboWidget.js]: > Other non content relevant keypresses also trigger autodrop (alt, numlock, context menu key, volume keys if any, insert key, pause key, etc.)

So it is not a good idea just add blacklist (try `print screen`)

> Possible improvement could be to use an onInput or onChange

use these events to open might be better

**Comment by scribetw** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/zkoss/zk/1123)*

File: zktest/src/archive/test2/B85-ZK-3702.zul [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        History:
+                Wed May 30 17:40:36 CST 2018, Created by klyve
+
+Copyright (C) 2018 Potix Corporation. All Rights Reserved.
+
+-->
+
+<zk>
+	<label multiline="true">
+		1. click the combobox and shoud focus on the combobox.
```
[the remark this answers is not in the document]
Comment (klyvechen) [reply zktest/src/archive/test2/B85-ZK-3702.zul]: ok

File: zul/src/archive/web/js/zul/inp/ComboWidget.js (hunk 1 above)
Comment (klyvechen) [reply zul/src/archive/web/js/zul/inp/ComboWidget.js]: ok

File: zul/src/archive/web/js/zul/inp/ComboWidget.js [hunk 3]
```diff
@@ -522,6 +527,8 @@ zul.inp.ComboWidget = zk.$extends(zul.inp.InputWidget, {
 		zWatch.unlisten({onSize: this, onFloatUp: this, onResponse: this, onScroll: this});
 		if (!zk.css3) jq.unzsync(this);
 
+		this.domUnlisten_(real, 'onInput');
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/inp/ComboWidget.js]: Where is `real` declared?

File: zul/src/archive/web/js/zul/inp/ComboWidget.js [hunk 4]
```diff
@@ -500,13 +500,18 @@ zul.inp.ComboWidget = zk.$extends(zul.inp.InputWidget, {
 	},
 	bind_: function () {
 		this.$supers(zul.inp.ComboWidget, 'bind_', arguments);
-		var btn, inp = this.getInputNode();
+		var btn, inp = this.getInputNode(), wgt = this;
```
Comment (wenninghsu) [inline zul/src/archive/web/js/zul/inp/ComboWidget.js]: separate to three lines

Comment (klyvechen) [reply zul/src/archive/web/js/zul/inp/ComboWidget.js]: ok

**Comment by wenninghsu** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**